### PR TITLE
feat/SubHeader: adding dropdown content to primary button

### PR DESCRIPTION
**Pull request by otaviolbarbosa**
image

## Summary by CodeRabbit

- **New Features**
	- Enhanced the header component with a dropdown feature for selecting actions like save and more.
	- Added a new exported entity `SubHeaderDropdownElement` for subheader with dropdown functionality.

### CODE REVIEW

File: packages/@orchidui-vue/src/Elements/Header/OcHeaderRight.vue [hunk 1]
```diff
@@ -22,7 +36,33 @@ defineProps({
...
+            class="min-w-[100px]"
+            label="Save"
+            v-bind="primaryProps"
+            is-additional-area
+            additional-area-icon="chevron-down"
+            @click="$emit('save')"
+            @addition-click="isDropdownOpen = true"
+          />
+          <template #menu>
+            <div class="flex flex-col">
```
Comment (HoaiLee) [inline packages/@orchidui-vue/src/Elements/Header/OcHeaderRight.vue:50]: I think it's safe to remove
```suggestion
```

File: packages/@orchidui-vue/src/Elements/Header/OcHeaderRight.vue [hunk 2]
```diff
@@ -22,7 +36,33 @@ defineProps({
           v-bind="secondaryProps"
           @click="$emit('cancel')"
         />
+        <Dropdown v-if="dropdownOptions?.length" v-model="isDropdownOpen" placement="bottom-end">
```
Comment (HoaiLee) [inline packages/@orchidui-vue/src/Elements/Header/OcHeaderRight.vue:39]: can use `slot` int this case?

**Review by HoaiLee — CHANGES_REQUESTED** [review]

File: packages/@orchidui-vue/src/Elements/Header/OcHeaderRight.vue (hunk 2 above)
Comment (otaviolbarbosa) [reply packages/@orchidui-vue/src/Elements/Header/OcHeaderRight.vue:39]: Yes, I can do it :)

Comment (otaviolbarbosa) [reply packages/@orchidui-vue/src/Elements/Header/OcHeaderRight.vue]: Done.

File: packages/@orchidui-vue/src/Elements/Header/OcHeaderRight.vue (hunk 1 above)
Comment (otaviolbarbosa) [reply packages/@orchidui-vue/src/Elements/Header/OcHeaderRight.vue]: Removed tis piece of code

File: packages/vue/src/Elements/Header/OcHeaderRight.vue [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <script setup>
 import { Button } from "@/orchidui";
-defineEmits(["save", "cancel"]);
+const emit = defineEmits(["save", "cancel", "addition-click", "dropdown-action"]);
```
Comment (HoaiLee) [inline packages/vue/src/Elements/Header/OcHeaderRight.vue:3]: Please remove unusage event

**Pull request merged**